// react-lightbox-gallery/dist/index.js
'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css = ".styles_row__2gmnz {\n  column-count: 4;\n  transition: all .5s ease-in-out;\n  column-gap: 10px;\n  column-fill: initial;\n}\n\n/* Create four equal columns that sits next to each other */\n\n.styles_column__2r45X {\n  margin-bottom: 10px;\n  display: inline-block;\n  vertical-align: top;\n}\n\n.styles_column__2r45X img {\n  vertical-align: middle;\n  width: 100%;\n}\n\n.styles_lightroom__1X2qE {\n  height: 100vh;\n  width: 100vw;\n  position: absolute;\n  z-index: 20;\n  top: 0;\n  left: 0;\n  background-color: #fff;\n  overflow-x: hidden;\n  transition: all 0.15s;\n  visibility: hidden;\n}\n\n.styles_carouselcontrolprev__Bmyua {\n  position: absolute;\n  left: 5%;\n  top: 45%;\n}\n\n.styles_carouselcontrolnext__DoQ9- {\n  position: absolute;\n  right: 5%;\n  top: 45%;\n}\n\n.styles_lightroomcontent__1SCaZ {\n  margin: auto;\n  height: 80%;\n  width: 80vw;\n  text-align: center;\n  display: flex;\n  align-items: center;\n  justify-content: center;\n}\n\n.styles_lightroomdesc__rNPED {\n  margin:auto;\n  height: 10%;\n  text-align: center;\n  width: 80vw;\n}\n\n.styles_lightroomimg__1r0lF {\n  max-width: 100%;\n  max-height: 80%;\n}\n\n.styles_topmenu__oGT_u {\n  position: absolute;\n  right: 5%;\n  top:1%;\n  font-size: medium;\n}\n.styles_sub__363ZQ{\n  font-style: italic;\n  font-size: 14px;\n}\n.styles_icon__1uWQb{\n  margin-left:15px;\n  font-size: 20px;\n}\n.styles_thumbpanel__1sa4E{\n  height:100vh;\n  width:0vw;\n  background: #fff;\n  position: fixed;\n  z-index: 20;\n  top: 0;\n  right: 0;\n  transition: 0.15s;\n  overflow-y:auto ;\n  padding-bottom: 10px;\n  text-align: center;\n}\n.styles_thumbnail__19ffY{\n  height:12vh;\n  margin-top:10px;\n  width:80%;\n  background: center center;\n  background-size: cover;\n  margin-left: auto;\n  margin-right:auto;\n}\n/* Responsive layout - makes a two column-layout instead of four columns */\n\n@media screen and (max-width: 800px) {\n  .styles_row__2gmnz {\n    column-count: 4 ;\n  }\n  .styles_carouselcontrolprev__Bmyua {\n    left: 4%;\n  }\n  .styles_carouselcontrolnext__DoQ9- {\n    right: 4%;\n  }\n}\n\n/* Responsive layout - makes the two columns stack on top of each other instead of next to each other */\n\n@media screen and (max-width: 600px) {\n  .styles_row__2gmnz {\n    column-count: 2 ;\n  }\n  .styles_carouselcontrolprev__Bmyua {\n    left: 3%;\n  }\n  .styles_carouselcontrolnext__DoQ9- {\n    right: 3%;\n  }\n}";
var styles = { "row": "styles_row__2gmnz", "column": "styles_column__2r45X", "lightroom": "styles_lightroom__1X2qE", "carouselcontrolprev": "styles_carouselcontrolprev__Bmyua", "carouselcontrolnext": "styles_carouselcontrolnext__DoQ9-", "lightroomcontent": "styles_lightroomcontent__1SCaZ", "lightroomdesc": "styles_lightroomdesc__rNPED", "lightroomimg": "styles_lightroomimg__1r0lF", "topmenu": "styles_topmenu__oGT_u", "sub": "styles_sub__363ZQ", "icon": "styles_icon__1uWQb", "thumbpanel": "styles_thumbpanel__1sa4E", "thumbnail": "styles_thumbnail__19ffY" };
styleInject(css);

var _ref =
/*#__PURE__*/
React__default.createElement("path", {
  d: "M18 6L6 18M6 6l12 12"
});

var closeIconLight = "data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%2224%22%20height%3D%2224%22%20viewBox%3D%220%200%2024%2024%22%20fill%3D%22none%22%20stroke%3D%22%23ffffff%22%20stroke-width%3D%222%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%20class%3D%22feather%20feather-x%22%3E%3Cline%20x1%3D%2218%22%20y1%3D%226%22%20x2%3D%226%22%20y2%3D%2218%22%3E%3C%2Fline%3E%3Cline%20x1%3D%226%22%20y1%3D%226%22%20x2%3D%2218%22%20y2%3D%2218%22%3E%3C%2Fline%3E%3C%2Fsvg%3E";

var _ref$1 =
/*#__PURE__*/
React__default.createElement("path", {
  d: "M18 6L6 18M6 6l12 12"
});

var closeIconDark = "data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%2224%22%20height%3D%2224%22%20viewBox%3D%220%200%2024%2024%22%20fill%3D%22none%22%20stroke%3D%22currentColor%22%20stroke-width%3D%222%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%20class%3D%22feather%20feather-x%22%3E%3Cline%20x1%3D%2218%22%20y1%3D%226%22%20x2%3D%226%22%20y2%3D%2218%22%3E%3C%2Fline%3E%3Cline%20x1%3D%226%22%20y1%3D%226%22%20x2%3D%2218%22%20y2%3D%2218%22%3E%3C%2Fline%3E%3C%2Fsvg%3E";

var _ref$2 =
/*#__PURE__*/
React__default.createElement("path", {
  d: "M21 10H3M21 6H3M21 14H3M21 18H3"
});

var thumbnailLight = "data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%2224%22%20height%3D%2224%22%20viewBox%3D%220%200%2024%2024%22%20fill%3D%22none%22%20stroke%3D%22%23ffffff%22%20stroke-width%3D%222%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%20class%3D%22feather%20feather-align-justify%22%3E%3Cline%20x1%3D%2221%22%20y1%3D%2210%22%20x2%3D%223%22%20y2%3D%2210%22%3E%3C%2Fline%3E%3Cline%20x1%3D%2221%22%20y1%3D%226%22%20x2%3D%223%22%20y2%3D%226%22%3E%3C%2Fline%3E%3Cline%20x1%3D%2221%22%20y1%3D%2214%22%20x2%3D%223%22%20y2%3D%2214%22%3E%3C%2Fline%3E%3Cline%20x1%3D%2221%22%20y1%3D%2218%22%20x2%3D%223%22%20y2%3D%2218%22%3E%3C%2Fline%3E%3C%2Fsvg%3E";

var _ref$3 =
/*#__PURE__*/
React__default.createElement("path", {
  d: "M21 10H3M21 6H3M21 14H3M21 18H3"
});

var thumbnailDark = "data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%2224%22%20height%3D%2224%22%20viewBox%3D%220%200%2024%2024%22%20fill%3D%22none%22%20stroke%3D%22currentColor%22%20stroke-width%3D%222%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%20class%3D%22feather%20feather-align-justify%22%3E%3Cline%20x1%3D%2221%22%20y1%3D%2210%22%20x2%3D%223%22%20y2%3D%2210%22%3E%3C%2Fline%3E%3Cline%20x1%3D%2221%22%20y1%3D%226%22%20x2%3D%223%22%20y2%3D%226%22%3E%3C%2Fline%3E%3Cline%20x1%3D%2221%22%20y1%3D%2214%22%20x2%3D%223%22%20y2%3D%2214%22%3E%3C%2Fline%3E%3Cline%20x1%3D%2221%22%20y1%3D%2218%22%20x2%3D%223%22%20y2%3D%2218%22%3E%3C%2Fline%3E%3C%2Fsvg%3E";

var _ref$4 =
/*#__PURE__*/
React__default.createElement("path", {
  d: "M19 12H5M12 19l-7-7 7-7"
});

var arrowLeftLight = "data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%2224%22%20height%3D%2224%22%20viewBox%3D%220%200%2024%2024%22%20fill%3D%22none%22%20stroke%3D%22%23ffffff%22%20stroke-width%3D%222%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%20class%3D%22feather%20feather-arrow-left%22%3E%3Cline%20x1%3D%2219%22%20y1%3D%2212%22%20x2%3D%225%22%20y2%3D%2212%22%3E%3C%2Fline%3E%3Cpolyline%20points%3D%2212%2019%205%2012%2012%205%22%3E%3C%2Fpolyline%3E%3C%2Fsvg%3E";

var _ref$5 =
/*#__PURE__*/
React__default.createElement("path", {
  d: "M19 12H5M12 19l-7-7 7-7"
});

var arrowLeftDark = "data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%2224%22%20height%3D%2224%22%20viewBox%3D%220%200%2024%2024%22%20fill%3D%22none%22%20stroke%3D%22currentColor%22%20stroke-width%3D%222%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%20class%3D%22feather%20feather-arrow-left%22%3E%3Cline%20x1%3D%2219%22%20y1%3D%2212%22%20x2%3D%225%22%20y2%3D%2212%22%3E%3C%2Fline%3E%3Cpolyline%20points%3D%2212%2019%205%2012%2012%205%22%3E%3C%2Fpolyline%3E%3C%2Fsvg%3E";

var _ref$6 =
/*#__PURE__*/
React__default.createElement("path", {
  d: "M5 12h14M12 5l7 7-7 7"
});

var arrowRightLight = "data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%2224%22%20height%3D%2224%22%20viewBox%3D%220%200%2024%2024%22%20fill%3D%22none%22%20stroke%3D%22%23ffffff%22%20stroke-width%3D%222%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%20class%3D%22feather%20feather-arrow-right%22%3E%3Cline%20x1%3D%225%22%20y1%3D%2212%22%20x2%3D%2219%22%20y2%3D%2212%22%3E%3C%2Fline%3E%3Cpolyline%20points%3D%2212%205%2019%2012%2012%2019%22%3E%3C%2Fpolyline%3E%3C%2Fsvg%3E";

var _ref$7 =
/*#__PURE__*/
React__default.createElement("path", {
  d: "M5 12h14M12 5l7 7-7 7"
});

var arrowRightDark = "data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%2224%22%20height%3D%2224%22%20viewBox%3D%220%200%2024%2024%22%20fill%3D%22none%22%20stroke%3D%22currentColor%22%20stroke-width%3D%222%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%20class%3D%22feather%20feather-arrow-right%22%3E%3Cline%20x1%3D%225%22%20y1%3D%2212%22%20x2%3D%2219%22%20y2%3D%2212%22%3E%3C%2Fline%3E%3Cpolyline%20points%3D%2212%205%2019%2012%2012%2019%22%3E%3C%2Fpolyline%3E%3C%2Fsvg%3E";

var _typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol" ? function (obj) {
  return typeof obj;
} : function (obj) {
  return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj;
};

var classCallCheck = function (instance, Constructor) {
  if (!(instance instanceof Constructor)) {
    throw new TypeError("Cannot call a class as a function");
  }
};

var createClass = function () {
  function defineProperties(target, props) {
    for (var i = 0; i < props.length; i++) {
      var descriptor = props[i];
      descriptor.enumerable = descriptor.enumerable || false;
      descriptor.configurable = true;
      if ("value" in descriptor) descriptor.writable = true;
      Object.defineProperty(target, descriptor.key, descriptor);
    }
  }

  return function (Constructor, protoProps, staticProps) {
    if (protoProps) defineProperties(Constructor.prototype, protoProps);
    if (staticProps) defineProperties(Constructor, staticProps);
    return Constructor;
  };
}();

var _extends$8 = Object.assign || function (target) {
  for (var i = 1; i < arguments.length; i++) {
    var source = arguments[i];

    for (var key in source) {
      if (Object.prototype.hasOwnProperty.call(source, key)) {
        target[key] = source[key];
      }
    }
  }

  return target;
};

var inherits = function (subClass, superClass) {
  if (typeof superClass !== "function" && superClass !== null) {
    throw new TypeError("Super expression must either be null or a function, not " + typeof superClass);
  }

  subClass.prototype = Object.create(superClass && superClass.prototype, {
    constructor: {
      value: subClass,
      enumerable: false,
      writable: true,
      configurable: true
    }
  });
  if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass;
};

var possibleConstructorReturn = function (self, call) {
  if (!self) {
    throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
  }

  return call && (typeof call === "object" || typeof call === "function") ? call : self;
};

var LightRoomComponent = function (_Component) {
  inherits(LightRoomComponent, _Component);

  function LightRoomComponent() {
    var _ref;

    var _temp, _this, _ret;

    classCallCheck(this, LightRoomComponent);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = possibleConstructorReturn(this, (_ref = LightRoomComponent.__proto__ || Object.getPrototypeOf(LightRoomComponent)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
      length: 0,
      lightroomactive: false,
      thumbmenuactive: false,
      activeindex: 0,
      path: "",
      desc: "",
      sub: "",
      touchStart: 0,
      touchEnd: 0
    }, _this.size = 4, _this.mode = "light", _this.openlightroom = function (e) {
      _this.setState({
        lightroomactive: true
      });
      _this.getcontent(e.target.getAttribute("data-index"));
    }, _this.closelightroom = function () {
      _this.setState({
        lightroomactive: false,
        thumbmenuactive: false
      });
    }, _this.moveright = function () {
      var id = parseInt(_this.state.activeindex, 10);
      if (id == _this.state.length - 1) _this.setState({
        lightroomactive: false,
        thumbmenuactive: false
      });else _this.getcontent(++id);
    }, _this.moveleft = function () {
      var id = parseInt(_this.state.activeindex, 10);
      if (id == 0) _this.setState({
        lightroomactive: false,
        thumbmenuactive: false
      });else _this.getcontent(--id);
    }, _this.getcontent = function (id) {
      _this.setState({
        activeindex: id,
        path: _this.props.images[id].src,
        sub: _this.props.images[id].sub,
        desc: _this.props.images[id].desc
      });
    }, _this.thumbmenutoggle = function () {
      _this.setState({
        thumbmenuactive: !_this.state.thumbmenuactive
      });
    }, _this.calculateStyles = function () {
      var columncount = 4,
          lightroomBackground = "rgba(255,255,255,0.95)",
          textColor = "#000";
      if (_this.props.settings) {
        if (_this.props.settings.columnCount && (typeof window === 'undefined' ? 'undefined' : _typeof(window)) != undefined) {
          if (window.outerWidth <= 600) columncount = _this.props.settings.columnCount.mobile ? _this.props.settings.columnCount.mobile : 2;else if (window.outerWidth <= 800) columncount = _this.props.settings.columnCount.tab ? _this.props.settings.columnCount.tab : 3;else columncount = _this.props.settings.columnCount.default ? _this.props.settings.columnCount.default : 5;
        }
        lightroomBackground = _this.props.settings.mode == "light" ? "rgba(255,255,255,0.95)" : "rgba(0,0,0,0.95)";
        textColor = _this.props.settings.mode == "light" ? "#000" : "#fff";
        var lightroomwidth = _this.state.thumbmenuactive ? "85vw" : "100vw";
      }
      return {
        row: {
          columnCount: columncount
        },
        lightroom: {
          backgroundColor: lightroomBackground,
          color: textColor,
          width: lightroomwidth
        },
        thumbpanel: {
          width: _this.state.thumbmenuactive ? "15vw" : "0vw",
          background: _this.props.settings.mode == "light" ? "rgba(255,255,255,0.95)" : "rgba(0,0,0,0.95)"
        }
      };
    }, _this.touchstart = function (e) {
      _this.setState({
        touchStart: e.touches[0].clientX
      });
    }, _this.touchmove = function (e) {
      _this.setState({
        touchEnd: e.touches[0].clientX
      });
    }, _this.touchend = function (e) {
      var _this$state = _this.state,
          touchStart = _this$state.touchStart,
          touchEnd = _this$state.touchEnd;

      if (touchStart - touchEnd > 150) _this.moveright();
      if (touchStart - touchEnd < -150) _this.moveleft();
    }, _temp), possibleConstructorReturn(_this, _ret);
  }
  // intial state

  // varibles to store props


  createClass(LightRoomComponent, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      this.setState({
        length: this.props.images.length
      });
    }
    // controlling lightbox activation


    //lightbox controls

    // get image at current active index

    // toggle thumb menu

    // Calculate styles - reponsible for changing style based on settings

    // Swipe Controls

  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      var runtimeStyles = this.calculateStyles();

      return React__default.createElement(
        'div',
        null,
        React__default.createElement(
          'div',
          { className: styles.row, style: runtimeStyles.row },
          this.props.images.map(function (img, i) {
            return React__default.createElement(
              'div',
              { className: styles.column, key: i },
              React__default.createElement('img', {
                'data-index': i,
                src: img.src,
                onClick: _this2.openlightroom
              })
            );
          })
        ),
        React__default.createElement(
          'div',
          {
            className: styles.lightroom,
            style: _extends$8({
              visibility: this.state.lightroomactive ? "visible" : "hidden"
            }, runtimeStyles.lightroom)
          },
          React__default.createElement(
            'div',
            { className: styles.topmenu },
            React__default.createElement('img', {
              src: this.props.settings.mode == "light" ? thumbnailDark : thumbnailLight,
              className: styles.icon,
              onClick: this.thumbmenutoggle
            }),
            React__default.createElement('img', {
              src: this.props.settings.mode == "light" ? closeIconDark : closeIconLight,
              className: styles.icon,
              onClick: this.closelightroom
            })
          ),
          React__default.createElement(
            'div',
            {
              className: styles.lightroomcontent,
              onTouchStart: this.touchstart,
              onTouchMove: this.touchmove,
              onTouchEnd: this.touchend
            },
            React__default.createElement('img', {
              className: styles.lightroomimg,
              src: this.state.path,
              style: { maxWidth: "100%" }
            })
          ),
          React__default.createElement(
            'div',
            { className: styles.lightroomdesc },
            React__default.createElement(
              'h1',
              null,
              this.state.name
            ),
            React__default.createElement(
              'p',
              { className: styles.desc },
              this.state.desc
            ),
            React__default.createElement(
              'p',
              { className: styles.sub },
              this.state.sub
            )
          ),
          React__default.createElement('img', {
            src: this.props.settings.mode == "light" ? arrowLeftDark : arrowLeftLight,
            className: styles.carouselcontrolprev,
            role: 'button',
            onClick: this.moveleft
          }),
          React__default.createElement('img', {
            src: this.props.settings.mode == "light" ? arrowRightDark : arrowRightLight,
            className: styles.carouselcontrolnext,
            role: 'button',
            onClick: this.moveright
          })
        ),
        React__default.createElement(
          'div',
          {
            className: styles.thumbpanel,
            style: runtimeStyles.thumbpanel
          },
          this.props.images.map(function (img, i) {
            return React__default.createElement('div', {
              className: styles.thumbnail,
              'data-index': i,
              key: i,
              onClick: _this2.openlightroom,
              style: {
                backgroundImage: 'url(' + img.src + ')'
              }
            });
          })
        )
      );
    }
  }]);
  return LightRoomComponent;
}(React.Component);

module.exports = LightRoomComponent;
